### PR TITLE
mds: Reorganize class members in Server header

### PR DESCRIPTION
**Pull request by varshar16**
Fixes: https://tracker.ceph.com/issues/42867
Signed-off-by: Varsha Rao <varao@redhat.com>

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/mds/Server.h [hunk 1]
```diff
@@ -349,11 +315,43 @@ class Server {
...
   void flush_session(Session *session, MDSGatherBuilder *gather);
   void clear_batch_ops(const MDRequestRef& mdr);
 
+  MDSRank *mds;
+  MDCache *mdcache;
+  MDLog *mdlog;
+  PerfCounters *logger;
+
+  // OSDMap full status, used to generate ENOSPC on some operations
+  bool is_full;
```
Comment (batrick) [inline src/mds/Server.h]: `= false`!

File: src/mds/Server.h [hunk 2]
```diff
@@ -349,11 +315,43 @@ class Server {
...
+
+  // State for while in reconnect
+  MDSContext *reconnect_done;
```
Comment (batrick) [inline src/mds/Server.h]: `= nullptr`!

File: src/mds/Server.h [hunk 3]
```diff
@@ -349,11 +315,43 @@ class Server {
...
+  int failed_reconnects;
```
Comment (batrick) [inline src/mds/Server.h]: `= 0;`

File: src/mds/Server.h [hunk 4]
```diff
@@ -349,11 +315,43 @@ class Server {
...
+  bool reconnect_evicting;  // true if I am waiting for evictions to complete
```
Comment (batrick) [inline src/mds/Server.h]: `false`

File: src/mds/Server.h [hunk 5]
```diff
@@ -349,11 +315,43 @@ class Server {
   void evict_cap_revoke_non_responders();
   void handle_conf_change(const std::set<std::string>& changed);
 
+  bool terminating_sessions;
```
Comment (batrick) [inline src/mds/Server.h]: `false`

File: src/mds/Server.h (hunk 1 above)
Comment (varshar16) [reply src/mds/Server.h]: Done.

File: src/mds/Server.h (hunk 3 above)
Comment (varshar16) [reply src/mds/Server.h]: Done.

File: src/mds/Server.h (hunk 2 above)
Comment (varshar16) [reply src/mds/Server.h]: Done.

File: src/mds/Server.h (hunk 5 above)
Comment (varshar16) [reply src/mds/Server.h]: Done.

File: src/mds/Server.h (hunk 4 above)
Comment (varshar16) [reply src/mds/Server.h]: Done.

**Comment by varshar16** [comment]
jenkins retest this please

**Comment by varshar16** [comment]
Rebased.

**Pull request merged**